### PR TITLE
[BE] Edit job

### PR DESCRIPTION
**Pull request by MihaiDruta**
[BE] Edit job
https://trello.com/c/Cb1o51VT/12-be-edit-job

Added necessary functions for editing an activity.

### CODE REVIEW

**Review by nspc97 — APPROVED** [review]

File: src/Controller/ActivityController.php [hunk 1]
```diff
@@ -157,4 +157,61 @@ public function createAction(Request $request): Response
...
+                [
+                    'message' => $exception->getMessage(),
+                    'entity' => $exception->getEntity(),
+                    'id' => $exception->getId()
+                ],
+                Response::HTTP_NOT_FOUND
+            );
+        }
+
+        $manager->persist($activityToEdit);
```
Comment (SpartakusMd) [inline src/Controller/ActivityController.php]: Implement a save method in the repository instead of this.

File: src/Entity/Activity.php [hunk 2]
```diff
@@ -110,6 +110,16 @@ public function __construct()
         $this->updatedAt = new DateTime();
     }
 
+    public function getTechnologies()
```
Comment (SpartakusMd) [inline src/Entity/Activity.php]: Add return type + DocBlock (comment)

File: src/Entity/Activity.php [hunk 3]
```diff
@@ -110,6 +110,16 @@ public function __construct()
...
+    {
+        return $this->technologies;
+    }
+
+    public function getTypes()
```
Comment (SpartakusMd) [inline src/Entity/Activity.php]: Add return type + DocBlock (comment)

File: src/Service/ActivityTransformer.php [hunk 4]
```diff
@@ -49,7 +54,25 @@ public function transform(
         ActivityDTO $dto
     ): Activity {
```
Comment (SpartakusMd) [inline src/Service/ActivityTransformer.php:115]: This method has become too big. Please separate into multiple private methods.

File: src/Controller/ActivityController.php [hunk 5]
```diff
@@ -157,4 +157,61 @@ public function createAction(Request $request): Response
...
+            ActivityDTO::class,
+            'json',
+            $context
+        );
+
+        $errors = $this->validator->validate($activityDTO);
+
+        if (count($errors) > 0) {
+            $errorsString = (string)$errors;
+            return new Response($errorsString);
```
Comment (SpartakusMd) [inline src/Controller/ActivityController.php]: Maybe JsonResponse with the status code 400?

**Review by SpartakusMd — CHANGES_REQUESTED** [review]

**Review by SpartakusMd — APPROVED** [review]

**Pull request merged**